src/components/verification-grid/verification-grid.ts

```diff
@@ -866,72 +891,33 @@ export class VerificationGridComponent extends AbstractComponent(LitElement) {
 
   private async handlePreviousPageClick(): Promise<void> {
     if (this.canNavigatePrevious()) {
-      this.historyHead += this.populatedTileCount;
-      await this.renderHistory(this.historyHead);
+      this.pageBackward();
     }
   }
 
   private handleNextPageClick(): void {
-    if (this.isViewingHistory()) {
-      this.pageForwardHistory();
-    }
-  }
-
-  private async pageForwardHistory(): Promise<void> {
     if (this.canNavigateNext()) {
-      this.historyHead -= this.populatedTileCount;
-      await this.renderHistory(this.historyHead);
+      this.pageForward();
     }
   }
 
-  private async renderHistory(historyOffset: number) {
-    const decisionStart = Math.max(0, this.subjectHistory.length - historyOffset);
-    const decisionEnd = Math.min(this.subjectHistory.length, decisionStart + this.populatedTileCount);
-    const decisionHistory = this.subjectHistory.slice(decisionStart, decisionEnd);
+  private async pageForward(): Promise<void> {
+    const proposedViewHead = this.viewHead + this.populatedTileCount;
+    this.viewHead += Math.min(proposedViewHead, this.verificationHead);
```

Comment:
possible overflow of subjects array?

or are just counting on the button being disabled so that this isn't called at the end?

Reply:
Correct, this is potential overflow of the subjects array

I can add a guard to the `viewHead` setter that will clamp it to the length of the `subjects` array

It will also emit a warning if there is an attempt to set the `viewHead` to a value larger than the subject array's length